lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -856,6 +856,89 @@ SwiftLanguage::GetHardcodedSynthetics() {
           }
           return nullptr;
         });
+    g_formatters.push_back([](lldb_private::ValueObject &valobj,
+                              lldb::DynamicValueType dyn_type,
+                              FormatManager &format_manager)
+                               -> lldb::SyntheticChildrenSP {
+      // Try to recognize a Swift type imported in C++. These types have a
+      // static constexpr char field whose name is the type's Swift mangled
+      // name.
+      ProcessSP process_sp(valobj.GetProcessSP());
+      auto *swift_runtime = SwiftLanguageRuntime::Get(process_sp);
+      if (!swift_runtime)
+        return nullptr;
+
+      auto module = valobj.GetModule();
+      if (!module)
+        return nullptr;
+
+      auto gts = module->GetTypeSystemForLanguage(
+          lldb::LanguageType::eLanguageTypeSwift);
+      if (!gts)
+        return nullptr;
+
+      auto *ts = llvm::cast<TypeSystemSwift>(gts->get());
+
+      // This only makes sense for Clang types.
+      auto type = valobj.GetCompilerType();
+      auto tsc = type.GetTypeSystem().dyn_cast_or_null<TypeSystemClang>();
+      if (!tsc)
+        return nullptr;
+
+      // We operate directly on the qualifier type because the TypeSystem
+      // interface doesn't allow us to check for static constexpr members.
+      auto qual_type = TypeSystemClang::GetQualType(type.GetOpaqueQualType());
+      const clang::Type::TypeClass type_class = qual_type->getTypeClass();
+      // If this a C++ wrapper class, it's definitely a record.
+      if (type_class != clang::Type::Record)
+        return nullptr;
+
+      auto *record_type = llvm::cast<clang::RecordType>(qual_type.getTypePtr());
+      const clang::RecordDecl *record_decl = record_type->getDecl();
+      CompilerType swift_type;
+      for (auto *child_decl : record_decl->decls()) {
+        if (auto *var_decl = llvm::dyn_cast<clang::VarDecl>(child_decl)) {
+          auto name = var_decl->getName();
+          // If the name of the var decl is a valid swift mangled name, then
+          // this is the Swift type.
+          if (swift::Demangle::isMangledName(name)) {
```

Comment:
Thinking of it, this is kind of slow. Should we consider using well-known name, and its type contains the mangled name, so we don't need to do a linear search here and call the demangler n times?